### ISSUE TITLE
Trouble attaching to modified OpenJDK

### ISSUE THREAD

**Issue by WhatATopic** [apangin/jattach#15]
Not specifically a jattach issue but I really wanted to get in contact with you because you seem to have a good understanding of the Attach API and I could really use some help.

I am trying to attach to a java process and dump the classes at runtime. The java process uses a modified OpenJDK which causes [this error](https://gyazo.com/d962a979206fce34d3c8a9c135a78d46) every time I try to attach using VirtualMachine.attach(). I tried using your jattach and I get "Could not allocate memory in target process error code 5." I am also not sure if they are disabling the Attach API via command line arguments because the args are hidden. If you have any ideas how I could attach to this process, I would greatly appreciate it.

**Comment by apangin** [comment]
Error code 5 means Access Denied. Try to run jattach as Administrator.

**Comment by apangin** [comment]
What does `jinfo <pid>` say?

**Comment by WhatATopic** [comment]
It turns out they are using -XX:+DisableAttachMechanism as one of their JVM args. I ran jattach as admin and I still get the same "could not allocate memory in target process error code 5." Doing jinfo presents [this error](https://gyazo.com/1a8af74c3fe0bc55fb8f0f90cdf6a201). I'm assuming because they are disabling the attach mechanism, I am fucked unless I can remove the argument from the native launcher exe that is applying it.

**Comment by apangin** [comment]
Right, `DisableAttachMechanism` prevents from starting JVM Attach Listener thread, so that `VirtualMachine.attach()` does not work.

In theory, a process with administrator privileges could inject into the JVM and switch `DisableAttachMechanism` off.
